app/models/measurement.rb

```diff
@@ -52,4 +52,64 @@ def self.find_by_perm_link(perm_link)
     perm_link = perm_link.sub('lmi_total_custom_', '').sub('lmi_total_', '')
     find_by(perm_link: ["lmi_total_#{perm_link}", "lmi_total_custom_#{perm_link}"])
   end
+
+  def load_gr_value(params)
```

Comment:
This goes back to our discussion from earlier, but how would you feel about making `load_gr_value` and its associated private methods moved to a new class, e.g. `Measurement::GrValueLoader` or something like that?